hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -190,6 +190,20 @@ default boolean closeIsCooperative() {
     default void close(@Nullable Throwable error) throws Exception {
     }
 
+    /**
+     * Returns {@code true} if this instance is reusable, i.e. {@link #init}
```

Comment:
this applies to `get` also

Reply:
I think we can elaborate more what reusability means, eg. that such PMS should not have any mutable state or side effects, allocate system resources, connections etc.

Reply:
I think we should just inform the implementor about the properties of reusable processor meta-suppliers, and let them decide whether the side effects, e.g. resource allocations, are tolerable.